robocorp-code/vscode-client/src/viewsRobotSelection.ts

```diff
@@ -0,0 +1,140 @@
+import * as vscode from 'vscode';
+import { basename, dirname } from 'path';
+
+import { debounce, FSEntry, getSelectedRobot, RobotEntry, treeViewIdToTreeView } from './viewsCommon';
+import { TREE_VIEW_ROBOCORP_ROBOTS_TREE } from './robocorpViews';
+
+export async function getCurrRobotDir(): Promise<FSEntry | undefined> {
+  let robotContentTree = treeViewIdToTreeView.get(TREE_VIEW_ROBOCORP_ROBOTS_TREE);
+  if (!robotContentTree) {
+      return undefined;
+  }
+
+  let parentEntry: FSEntry | undefined = undefined;
+  let selection: FSEntry[] = robotContentTree.selection;
+  if (selection.length > 0) {
+      parentEntry = selection[0];
+      if (!parentEntry.filePath) {
+          parentEntry = undefined;
+      }
+  }
+  if (!parentEntry) {
+      let robot: RobotEntry | undefined = getSelectedRobot();
+      if (!robot) {
+          return undefined;
+      }
+      parentEntry = {
+          filePath: dirname(robot.uri.fsPath),
+          isDirectory: true,
+          name: basename(robot.uri.fsPath)
+      }
+  }
+
+  if (!parentEntry.isDirectory) {
+      parentEntry = {
+          filePath: dirname(parentEntry.filePath),
+          isDirectory: true,
+          name: basename(parentEntry.filePath)
+      }
+  }
+
+  return parentEntry;
+}
+
+export class RobotSelectionTreeDataProviderBase implements vscode.TreeDataProvider<FSEntry> {
+
+  private _onDidChangeTreeData: vscode.EventEmitter<FSEntry | null> = new vscode.EventEmitter<FSEntry | null>();
+  readonly onDidChangeTreeData: vscode.Event<FSEntry | null> = this._onDidChangeTreeData.event;
+
+  protected lastRobotEntry: RobotEntry = undefined;
+  private lastWatcher: vscode.FileSystemWatcher | undefined = undefined;
+
+  fireRootChange() {
+      this._onDidChangeTreeData.fire(null);
+  }
+
+  robotSelectionChanged(robotEntry: RobotEntry) {
+      // When the robot selection changes, we need to start tracking file-changes at the proper place.
+      if (this.lastWatcher) {
+          this.lastWatcher.dispose();
+          this.lastWatcher = undefined;
+      }
+      this.fireRootChange();
+
+      let d = basename(dirname(robotEntry.uri.fsPath));
+      let watcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher('**/' + d + '/**', false, true, false);
```

Comment:
I know this is this way since it was already like that due to `RobotContentTreeDataProvider`, but after reviewing this API again, I think it should be something as:

`vscode.workspace.createFileSystemWatcher(new vscode.RelativePattern(robotEntry.uri, '**/**'), ...`

At this place we could potentially deal only with changes in the `devdata` folder for the case where we're dealing with work items (so, it should probably be customizable).